### ISSUE TITLE
verticalpodautoscaler CRD not compliant with openAPIV3

### ISSUE THREAD

**Issue by sebastien-prudhomme** [kubernetes/autoscaler#2859]
verticalpodautoscaler v0.6.3 with the patched CRD version from master branch (with some openAPIV3 fixes)

Just notice that there is still that error in the status of the verticalpodautoscaler CRD:

```
  - lastTransitionTime: "2020-02-26T10:44:51Z"
    message: 'spec.validation.openAPIV3Schema.type: Required value: must not be empty
      at the root'
    reason: Violations
    status: "True"
    type: NonStructuralSchema
```
With this patch, all is ok :

```diff
@ -24,6 +24,7 @@ spec:
       storage: true
   validation:
     openAPIV3Schema:
+      type: object
       properties:
         spec:
           type: object

```

**Comment by bskiba** [comment]
Thanks for reporting! Which Kubernetes version are you observing the error on?

**Comment by sebastien-prudhomme** [comment]
@bskiba Kubernetes v1.15.3

**Comment by onesolpark** [comment]
fixed by #2904
/close

**Comment by sebastien-prudhomme** [comment]
Thanks for the commit.
